File: lib/models/calendar_event.dart
enum EventType { homework, quiz, exam, project, other }

EventType _parseEventType(String? type) {
  switch (type) {
    case 'homework':
      return EventType.homework;
    case 'quiz':
      return EventType.quiz;
    case 'exam':
      return EventType.exam;
    case 'project':
      return EventType.project;
    default:
      return EventType.other;
  }
}

class CalendarEvent {
  final String id;
  final String courseId;
  final String title;
  final String description;
  final DateTime date;
  final EventType type;

  const CalendarEvent({
    required this.id,
    required this.courseId,
    required this.title,
    required this.description,
    required this.date,
    required this.type,
  });

  factory CalendarEvent.fromJson(Map<String, dynamic> json) {
    return CalendarEvent(
      id: json['id'] ?? json['_id'] ?? '',
      courseId: json['courseId'] ?? '',
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      date: DateTime.tryParse(json['date'] ?? '') ?? DateTime.now(),
      type: _parseEventType(json['type']),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'courseId': courseId,
    'title': title,
    'description': description,
    'date': date.toIso8601String(),
    'type': type.name,
  };
}

File: backend/models/CalendarEvent.js
const mongoose = require('mongoose');

const calendarEventSchema = new mongoose.Schema({
  courseId:    { type: String, required: true },
  title:       { type: String, required: true },
  description: { type: String, default: '' },
  date:        { type: Date, required: true },
  type:        { type: String, enum: ['homework', 'quiz', 'exam', 'project', 'other'], default: 'other' },
  createdBy:   { type: String },
}, { timestamps: true });

calendarEventSchema.methods.toJSON = function() {
  const obj = this.toObject();
  obj.id = obj._id.toString();
  delete obj._id;
  delete obj.__v;
  return obj;
};

calendarEventSchema.index({ courseId: 1, date: 1 });

module.exports = mongoose.model('CalendarEvent', calendarEventSchema);
